Expensify/App
# Update Desktop deploy to use npm ci and update packages

@AndrewGable @marcaaron will you please review this?

This PR fixes an issue that causes the desktop build to sometimes fail. The main fix here is to update `npm install` to `npm ci`, but to do that we needed to update a library that is linked in `expensify-common` to fix some broken syntax. Therefore, this PR contains the update to the desktop deploy workflow, as well as some package.json updates for the new library hashes.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/144368

### Tests
1. Run the app and sign out if you're already signed in
2. Confirm you can sign into the app
3. Confirm you can send messages to other accounts, and that the other accounts receive the messages
